js/SGBot.js

```diff
@@ -1,72 +1,80 @@
 'use strict';
 const Discord = require('discord.js');
-const bot = new Discord.Client();
+const utils = require('./utils.js');
 const config = require('../config.json');
 const credentials = require('../credentials.json');
-const utils = require('../js/utils.js');
 
-bot.login(credentials.token);
+const bot = new Discord.Client();
+bot.login(credentials.token).catch(err => {
+    console.error(err.message);
+    bot.destroy();
+});
 
-bot.on('ready', () => {
-    // bot.user.setUsername('Emoji Microwave (Temporary Name)')
+bot.on('ready', async () => {
+    try {
+        // await bot.user.setUsername('Emoji Microwave (Temporary Name)')
+        await bot.user.setPresence({ game: { name: 'with worldlines' }, status: 'online' });
+    } catch(err) {
+        console.error(err.stack);
+    }
     console.log('Awaiting D-mails!');
-    bot.user.setPresence({ game: { name: 'with worldlines' }, status: 'online' });
 });
 
-bot.on('message', (message) => {
-
+bot.on('message', async (message) => {
     // No-op if message was from a bot
     if(message.author.bot) return;
 
-    // String literal matches
-    if(message.content.toLowerCase() === 'nullpo') {
-        message.channel.send('Gah!');
-    }
+    try {
+        // String literal matches
+        if(message.content.toLowerCase() === 'nullpo') {
+            await message.channel.send('Gah!');
+        }
 
-    // Commands
-    if(message.content.startsWith(config.commandPrefix)) {
-        const command = utils.parseCommand(message.content);
-        switch (command.toLowerCase()) {
-            case 'help':
-                helpCommand(message);
-                break;
-            case 'tuturu':
-                playCommand(message, command);
-                break;
-            case 'beechga':
-                playCommand(message, command);
-                break;
-            case 'sonuvabetch':
-                playCommand(message, command);
-                break;
+        // Commands
+        if(message.content.startsWith(config.commandPrefix)) {
+            const command = utils.parseCommand(message.content);
+            switch(command.toLowerCase()) {
+                case 'help':
+                    await helpCommand(message);
+                    break;
+                case 'tuturu':
```

Comment:
Neat usage, I guess the "falling through" of cases when you don't `break;` can be useful